sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -879,89 +1492,103 @@ class CompactBlockProcessor internal constructor(
          * @param batch the batch of blocks to download.
          */
         @VisibleForTesting
-        @Throws(CompactBlockProcessorException.FailedDownload::class)
-        @Suppress("MagicNumber")
         internal suspend fun downloadBatchOfBlocks(
             downloader: CompactBlockDownloader,
             batch: BlockBatch
-        ): BlockProcessingResult {
+        ): SyncingResult {
             var downloadedBlocks = listOf<JniBlockMeta>()
-            retryUpTo(RETRIES, { CompactBlockProcessorException.FailedDownload(it) }) { failedAttempts ->
+            var downloadException: CompactBlockProcessorException.FailedDownloadException? = null
+
+            retryUpToAndContinue(
+                retries = RETRIES,
+                exceptionWrapper = {
+                    downloadException = CompactBlockProcessorException.FailedDownloadException(it)
+                    downloadException!!
+                }
+            ) { failedAttempts ->
+                @Suppress("MagicNumber")
```

Comment:
Where's the magic number here? (0 isn't magic.) Oh well, non-blocking.

Reply:
Good catch. Thanks! It's possibly a leftover from the previous iteration.